python_modules/dagster-graphql/dagster_graphql/implementation/loader.py

```diff
@@ -277,3 +284,111 @@ def get_latest_materialization_for_asset_key(self, asset_key: AssetKey) -> Event
     def _fetch(self):
         self._fetched = True
         self._materializations = self._instance.get_latest_materialization_events(self._asset_keys)
+
+
+class CrossRepoAssetDependencyLoader:
+    """
+    A batch loader that computes cross-repository asset dependencies. Locates source assets
+    within all workspace repositories, and determines if they are derived (defined) assets in
+    other repositories.
+
+    For each asset that contains cross-repo dependencies (every asset that is defined as a source
+    asset in another repository) a sink asset is any asset immediately downstream of the source
+    asset.
+
+    E.g. Asset A is defined in repo X and referenced in repo Y as source asset C (but contains the
+    same asset key as A). If within repo C has a downstream asset B, B is a sink asset of A (it
+    is external from A's repo but an edge exists from A to B).
+
+    """
+    def __init__(self, context: WorkspaceRequestContext):
+        self._context = context
+        self._fetched = False
```

Comment:
With the `lru_cache`, I think there's a way to structure the code that doesn't require `_fetched` anymore.  I.e. have `_build_cross_repo_deps` return the values that it creates instead of assigning them as class variables.